### PR TITLE
Added publish if user sent a freestyle/image solution (#207)

### LINKED ISSUE #207: Use PubSubService to publish if user has sent new freestyle/image solution

### PR DESCRIPTION
**Pull request by PAException**
Closes #207

### CODE REVIEW

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java [hunk 1]
```diff
@@ -52,10 +55,15 @@
 				optionalFreestyleSolution.get(),
 				dto.getSolution(),
 				UUID.randomUUID()
-			)
+				)
 		);
 
-		//TODO: PubSubService publish new Freestyle for teacher
+		this.messagePublisher.publish(new UserSentFreestyleSolutionDTO(
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java:70]: Not yet refined by concept

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java [hunk 2]
```diff
@@ -52,10 +55,15 @@
...
+				solutionResult.getResult().getTaskId().getAuthorId(),
+				UUID.randomUUID(),
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java]: Don't u have the user id?

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/ImageSolutionController.java [hunk 3]
```diff
@@ -51,10 +54,15 @@
 				optionalImageSolution.get(),
 				dto.getUrl(),
 				UUID.randomUUID()
-			)
+				)
 		);
 
-		//TODO: PubSubService publish new Image for teacher
+		this.messagePublisher.publish(new UserSentImageSolutionDTO(
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/solution/ImageSolutionController.java:67]: Not yet refined

**Review by steve-hb — CHANGES_REQUESTED** [review]

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java (hunk 2 above)
Comment (PAException) [reply src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java]: Not yet, would be given by a param of the method or in the dto

File: src/main/java/de/themorpheus/edu/taskservice/pubsub/dto/UserSentFreestyleSolutionDTO.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.stereotype.Component;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Component
+@PubSubDTO("task.task.solution.freestyle.user")
+public class UserSentFreestyleSolutionDTO {
+
+	private int taskId;
```
Comment (msentwickler) [inline src/main/java/de/themorpheus/edu/taskservice/pubsub/dto/UserSentFreestyleSolutionDTO.java:20]: Min(1) would be better

File: src/main/java/de/themorpheus/edu/taskservice/pubsub/dto/UserSentImageSolutionDTO.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.stereotype.Component;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Component
+@PubSubDTO("task.task.solution.image.user")
+public class UserSentImageSolutionDTO {
+
+	private int taskId;
```
Comment (msentwickler) [inline src/main/java/de/themorpheus/edu/taskservice/pubsub/dto/UserSentImageSolutionDTO.java:20]: same

**Review by msentwickler — CHANGES_REQUESTED** [review]

File: src/main/java/de/themorpheus/edu/taskservice/pubsub/dto/UserSentFreestyleSolutionDTO.java (hunk 4 above)
Comment (PAException) [reply src/main/java/de/themorpheus/edu/taskservice/pubsub/dto/UserSentFreestyleSolutionDTO.java:20]: not neccessary but why not

**Pull request merged**